File: hash.js
function hashSort(a) {
    let max = Math.max(...a);
    let table = new Array(max + 1).fill(0);

    for (let num of a)
        table[num] = 1;

    let sorted = [];
    for (let i = 0; i < table.length; i++)
        if (table[i] === 1)
            sorted.push(i);

    return sorted;
}

let a = [70, 15, 2, 51, 60];
console.log("Antes de ordenar:");
console.log(a.join(" "));
let sorted = hashSort(a);
console.log("Después de ordenar:");
console.log(sorted.join(" "));

File: hash.py
def hash_sort(a):
    max_val = max(a)
    table = [0] * (max_val + 1)

    for num in a:
        table[num] = 1

    sorted_a = []
    for i in range(len(table)):
        if table[i] == 1:
            sorted_a.append(i)

    return sorted_a

a = [70, 15, 2, 51, 60]
print("Antes de ordenar:", a)
a = hash_sort(a)
print("Después de ordenar:", a)
